### PR TITLE
[camera][ios] remove permission validation from init method

### LINKED ISSUE #11736: On iOS, Expo Camera requires NSMicrophoneUsageDescription key to use the feature
### Summary of Issue 
Hi, I just started using Expo Camera on my bare RN project and facing some issues.
I followed the installation process [here](https://github.com/expo/expo/tree/master/packages/expo-camera) and according to [Configure for iOS](https://github.com/expo/expo/tree/master/packages/expo-camera#configure-for-ios), I only need to add `NSCameraUsageDescription` key but doing so produces following error:
```
This app is missing either NSCameraUsageDescription or 
NSMicrophoneUsageDescription, so audio/video services will fail. Add one of
these entries to your bundle's Info.plist.
```
But adding `NSMicrophoneUsageDescription` key solves the error.
I simply copied the code from [here](https://docs.expo.io/versions/latest/sdk/camera/#usage) and I believe it shouldn't require permission for mic since I'm not going to use camera to take a video. So it should rather considered optional.

### Environment - output of `expo diagnostics` & the platform(s) you're targeting
```
Expo CLI 4.0.17 environment info:
    System:
      OS: macOS 11.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 15.4.0 - /usr/local/bin/node
      Yarn: 1.22.5 - ~/.yarn/bin/yarn
      npm: 7.0.15 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.10.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 14.3, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
    IDEs:
      Android Studio: 4.1 AI-201.8743.12.41.6953283
      Xcode: 12.3/12C33 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.13.1 => 16.13.1 
      react-native: 0.63.3 => 0.63.3 
    npmGlobalPackages:
      expo-cli: 4.0.17
    Expo Workflow: bare
```
 
### Steps to Reproduce
1. Create bare RN app with unimodules installed.
2. Install Expo Camera following this [page](https://github.com/expo/expo/tree/master/packages/expo-camera)
2. Copy the example from [here](https://docs.expo.io/versions/latest/sdk/camera/#usage)
3. Running it throws this error:
Simulator Screen Shot - iPhone 12 - 2021-01-25 at 17 35 43

### PR DESCRIPTION
**Pull request by ajsmth**
# Why

Possibly resolves #11736

I'm not sure how I missed this during QA, but there is an extra validation that checks for Plist entries in `initWithModuleRegistry` which is performing the same check that `requestPermissions` would do - e.g it validates that both camera and microphone permissions are in the target Info plist, and throws if they are not

Since we're expanding the API to pick and choose camera or microphone permissions, I don't think it makes sense to perform this check at all - any missing plist values will crash the app anyways, and if anything this method makes it more difficult to debug because the error message is clobbered by the validation, so it seems like the new APIs just aren't working

TLDR - If I install `expo-camera` in a bare app and use and include the camera permission APIs, this validation will still crash the app. 

# How

Removed the check in the init method

# Test Plan

Install in a bare ios app, include one permission value (microphone or camera),  and ensure that the app doesn't crash when using the Camera module permissions APIs - e.g `Camera.requestCameraPermissionAsync()`

### CODE REVIEW

File: packages/expo-camera/ios/EXCamera/EXCamera.m [hunk 1]
```diff
@@ -648,10 +648,6 @@ - (void)startSession
 #if TARGET_IPHONE_SIMULATOR
   return;
 #endif
-  if (![_permissionsManager hasGrantedPermissionUsingRequesterClass:[EXCameraPermissionRequester class]]) {
```
Comment (brentvatne) [inline packages/expo-camera/ios/EXCamera/EXCamera.m:651]: do we want to remove this entirely? it seems it would be better to check that the appropriate permissions are available still and continue to invoke [`onMountingError`](https://docs.expo.io/versions/latest/sdk/camera/#onmounterror) -- but i don't have full context on this

Comment (ajsmth) [reply packages/expo-camera/ios/EXCamera/EXCamera.m:651]: this check is performed on any call to `getPermissionsAsync()` but we don't know which ones the consuming app is going to require anymore, so making a check on mount no longer makes sense

Comment (brentvatne) [reply packages/expo-camera/ios/EXCamera/EXCamera.m:651]: @ajsmth - we know for sure that if they're trying to mount a camera that they need camera permissions, but we could defer check for recording audio until video recording is attempted (which i think would be handled by the aforementioned `getPermissionsAsync()`)

Comment (ajsmth) [reply packages/expo-camera/ios/EXCamera/EXCamera.m:651]: okay that's an easy change to make

Comment (brentvatne) [reply packages/expo-camera/ios/EXCamera/EXCamera.m:651]: the reason it's useful to keep this here is that otherwise the app may crash without a meaningful error message

**Review by brentvatne — APPROVED** [review]
